esphome/components/ams5915/ams5915.h

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/core/hal.h"
+#include "esphome/components/i2c/i2c.h"
+#include "esphome/components/sensor/sensor.h"
+
+namespace esphome {
+namespace ams5915 {
+enum Transducer {
+  AMS5915_0005_D,
+  AMS5915_0010_D,
+  AMS5915_0005_D_B,
+  AMS5915_0010_D_B,
+  AMS5915_0020_D,
+  AMS5915_0050_D,
+  AMS5915_0100_D,
+  AMS5915_0020_D_B,
+  AMS5915_0050_D_B,
+  AMS5915_0100_D_B,
+  AMS5915_0200_D,
+  AMS5915_0350_D,
+  AMS5915_1000_D,
+  AMS5915_2000_D,
+  AMS5915_4000_D,
+  AMS5915_7000_D,
+  AMS5915_10000_D,
+  AMS5915_0200_D_B,
+  AMS5915_0350_D_B,
+  AMS5915_1000_D_B,
+  AMS5915_1000_A,
+  AMS5915_1200_B
+};
+class Ams5915 : public PollingComponent, public sensor::Sensor, public i2c::I2CDevice {
+ public:
+  Ams5915() : PollingComponent(5000) {}
+  void dump_config() override;
+  void setup() override;
+  void update() override;
+  void set_temperature_sensor(sensor::Sensor *temperature_sensor) { temperature_sensor_ = temperature_sensor; }
```

Comment:
Please make sure to read and understand the [codebase standards](https://esphome.io/guides/contributing#codebase-standards)
```suggestion
  void set_temperature_sensor(sensor::Sensor *temperature_sensor) { this->temperature_sensor_ = temperature_sensor; }
```